packages/smart-wallet/src/smartWallet.js

```diff
@@ -366,6 +428,87 @@ export const prepareSmartWallet = (baggage, shared) => {
             },
           });
         },
+
+        provideBrandToPurses() {
+          const brandToPurses = provideLazy(
+            walletPurses,
+            this.facets.helper,
+            _k => {
+              /** @type {MapStore<Brand, PurseRecord[]>} */
+              const store = makeScalarBigMapStore('purses by brand', {
+                durable: true,
+              });
+              return store;
+            },
+          );
+          return brandToPurses;
+        },
+
+        /**
+         * Provide a purse given a NameHub of issuers and their
+         * brands.
+         *
+         * We current support only one NameHub, agoricNames, and
+         * hence one purse per brand. But we store an array of them
+         * to facilitate a transition decentralized introductions.
+         *
+         * @param {Brand} brand
+         * @param {ERef<NameHub>} known - namehub with brand, issuer branches
+         * @returns {Promise<Purse | undefined>} undefined if brand is not well known
+         */
+        async providePurseForKnownBrand(brand, known = shared.agoricNames) {
```

Comment:
ditto on abstracting provision

Reply:
The earlier case was more clear-cut, because there was no key argument. But here we do have a key -- the brand. If not here, where would we ever call something `provideX`? Perhaps only when we're also passing in a store?

Reply:
yeah, I think when the caller has the store in which it will be provided. if the caller just wants a thing and doesn't specify where to put it or how to make it, that's just a "get". Similar to how we don't put "memoize" in the name of functions that memoize.